### PR TITLE
:bug: normalization for bigquery: allow datasetId and table to start with number

### LINKED ISSUE #6732: Destination Bigquery: fail normalization when dataset_id start with number
## Enviroment
- **Airbyte version**:  0.30.3-alpha
- **OS Version / Instance**: macOS
- **Deployment**: Docker
- **Source Connector and version**: Hubspot
- **Destination Connector and version**: Bigquery 0.4.0
- **Severity**: Very Low / Low / Medium / **High** / Critical
- **Step where error happened**: Sync Job

## Current Behavior
One SQL generated file example, the dataset is wrong where `_111testing_raw` should be only `111testing_raw`.
```
select
    id,
    name,
    type,
    enabled,
    updatedAt,
    insertedAt,
    personaTagIds,
    contactListIds,
    _airbyte_emitted_at,
    _airbyte_workflows_hashid
from __dbt__CTE__workflows_ab3
-- workflows from `dataline-integration-testing`._111testing_raw._airbyte_raw_workflows
  );
```
## Expected Behavior
*Tell us what should happen.*

## Logs
[logs-3-0.txt](https://github.com/airbytehq/airbyte/files/7280851/logs-3-0.txt)

## Steps to Reproduce
1.
2.
3.

## Are you willing to submit a PR?

Remove this with your answer.

### PR DESCRIPTION
**Pull request by midavadim**
## What
If dataset_if started with a number like '1id' then normalization transformed it to '_1id'

REAL Example: 

select
    id,
    name,
    type,
    enabled,
    updatedAt,
    insertedAt,
    personaTagIds,
    contactListIds,
    _airbyte_emitted_at,
    _airbyte_workflows_hashid
from __dbt__CTE__workflows_ab3
-- workflows from `dataline-integration-testing`._111testing_raw._airbyte_raw_workflows

## How
Removed conditions in 
airbyte-integrations/bases/base-normalization/normalization/transform_catalog/destination_name_transformer.py

Manually tested that Bigquery allows the creation of dataset which starts with a number, 
Table name also can start with a number

## Recommended reading order
1. `x.java`
2. `y.python`

## 🚨 User Impact 🚨
No

## Pre-merge Checklist
Expand the relevant checklist and delete the others. 

<details><summary> <strong> New Connector </strong></summary>
<p>

#### Community member or Airbyter
   
- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/SUMMARY.md`
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
   
</p>
</details>

<details><summary> <strong> Updating a connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the new connector version is published, connector version bumped in the seed directory as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</p>
</details>

<details><summary> <strong> Connector Generator </strong> </summary>
<p>
   
- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed.
</p>
</details>

---
This change is [Reviewable](https://reviewable.io/reviews/airbytehq/airbyte/9341)

### CODE REVIEW

File: airbyte-integrations/bases/base-normalization/unit_tests/test_destination_name_transformer.py [hunk 1]
```diff
@@ -117,7 +117,7 @@ def test_transform_standard_naming(input_str: str, expected: str):
         ("a-Unicode_name_文6", "MSSQL", "a_unicode_name__6", "{{ adapter.quote('a-Unicode_name_文6') }}"),
         # Doesnt start with alpha or underscore
         ("100x2001", "Postgres", "100x2001", "{{ adapter.quote('100x2001') }}"),
-        ("100x2002", "BigQuery", "_100x2002", "_100x2002"),
```
Comment (bazarnov) [inline airbyte-integrations/bases/base-normalization/unit_tests/test_destination_name_transformer.py:120]: Should the input of "_100x2002" be transformed into "100x2002" for BIGQUERY specifically, based on this use case?

**Review by bazarnov — COMMENTED** [review]
Some other normalisation tests are not passing. Is it related to this particular change you're making? 
https://github.com/airbytehq/airbyte/runs/4774907109?check_suite_focus=true#step:10:20913

**Comment by midavadim** [comment]
1. current update is related to bigquery only

2. There is 1 failing test related to redshift test_nested_streams, it happens on the mater branch as well:


 >       assert error_count == 0, message
 E       AssertionError: docker run --rm --init -v /actions-runner/_work/airbyte/airbyte/airbyte-integrations/bases/base-normalization/integration_tests/normalization_test_output/redshift/test_nested_streams:/workspace -v /actions-runner/_work/airbyte/airbyte/airbyte-integrations/bases/base-normalization/integration_tests/normalization_test_output/redshift/test_nested_streams/build:/build -v /actions-runner/_work/airbyte/airbyte/airbyte-integrations/bases/base-normalization/integration_tests/normalization_test_output/redshift/test_nested_streams/logs:/logs -v /tmp:/tmp --network host --entrypoint /usr/local/bin/dbt -i airbyte/normalization:dev run --profiles-dir=/workspace --project-dir=/workspace --full-refresh
 E         	terminated with return code 1 with 1 'Error/Warning/Fail' mention(s).
 E       assert 1 == 0
 E         +1
 E         -0
 
 /actions-runner/_work/airbyte/airbyte/airbyte-integrations/bases/base-normalization/integration_tests/dbt_integration_test.py:465: AssertionError

**Comment by midavadim** [comment]
> Some other normalisation tests are not passing. Is it related to this particular change you're making? https://github.com/airbytehq/airbyte/runs/4774907109?check_suite_focus=true#step:10:20913

replied above

**Review by antixar — APPROVED** [review]
this is minor changes but tests were failed. I see their failures are related with another reason but we need approval of Airbyte team for publishing of your changes without tests.

**Comment by bazarnov** [comment]
@midavadim have you run the normalisation tests on `master` to see if it's failing or not without your changes?

**Review by bazarnov — APPROVED** [review]

**Comment by ChristopheDuong** [comment]
The docs says:

> Unquoted identifiers must begin with a letter or an underscore character. Subsequent characters can be letters, numbers, or underscores
> These are invalid identifiers:
> 5Customers


https://cloud.google.com/bigquery/docs/reference/standard-sql/lexical

**Comment by ChristopheDuong** [comment]
> Some other normalization tests are not passing. Is it related to this particular change you're making?

The normalization tests for redshift are failing indeed. 
I understand it is not related to this PR that is touching only BigQuery, but all tests should pass in order to merge...

Can we look into why redshift is failing too?

Was there any changes related to json/super types in redshift?
```
Completed with 1 error and 0 warnings:
	 
	 Database Error in model simple_stream_with_namespace_resulting_into_long_names (models/generated/airbyte_incremental/test_normalization_namespace/simple_stream_with_namespace_resulting_into_long_names.sql)
	   function json_extract_path_text(super, "unknown", boolean) does not exist
	   HINT:  No function matches the given name and argument types. You may need to add explicit type casts.
	   compiled SQL at ../build/run/airbyte_utils/models/generated/airbyte_incremental/test_normalization_namespace/simple_stream_with_namespace_resulting_into_long_names.sql
```

**Comment by midavadim** [comment]
> > Some other normalization tests are not passing. Is it related to this particular change you're making?
> 
> The normalization tests for redshift are failing indeed. I understand it is not related to this PR that is touching only BigQuery, but all tests should pass in order to merge...
> 
> Can we look into why redshift is failing too?
> 
> Was there any changes related to json/super types in redshift?
> 
> ```
> Completed with 1 error and 0 warnings:
> 	 
> 	 Database Error in model simple_stream_with_namespace_resulting_into_long_names (models/generated/airbyte_incremental/test_normalization_namespace/simple_stream_with_namespace_resulting_into_long_names.sql)
> 	   function json_extract_path_text(super, "unknown", boolean) does not exist
> 	   HINT:  No function matches the given name and argument types. You may need to add explicit type casts.
> 	   compiled SQL at ../build/run/airbyte_utils/models/generated/airbyte_incremental/test_normalization_namespace/simple_stream_with_namespace_resulting_into_long_names.sql
> ```

This problem was caused by overlapping tests from different branches.
Tests from another branch created tables with SUPER type, so that's why my tests failed.

Once we re-created redshift database my tests started working.

**Comment by midavadim** [comment]
> The official docs from Google BigQuery says:
> 
> > Unquoted identifiers must begin with a letter or an underscore character. Subsequent characters can be letters, numbers, or underscores
> > These are invalid identifiers:
> > 5Customers
> 
> https://cloud.google.com/bigquery/docs/reference/standard-sql/lexical
> 
> That's why it was implemented to insert an extra underscore
> 
> Why do we need to diverge from official recommendations? or do we need to adapt the logic for dataset and not table/column names?

I have tested different names on Bigquery:

Can not start with number: column (GCP also automatically adds _ to column names which starts with number during CSV upload)
Can start with number: datasetId, table (and they do not require quotes)
    SELECT name1, _2name, name_3 FROM `dataline-integration-testing.1.1tt` LIMIT 1000
    SELECT name1, _2name, name_3 FROM dataline-integration-testing.1.1tt LIMIT 1000


I tested only datasetId, table, columns ... is there another identifier that we should consider for big query?

**Review by ChristopheDuong — APPROVED** [review]
Great find thanks

**Pull request merged**